src/tool.cpp

```diff
@@ -153,36 +167,46 @@ void ToolCapabilities::deserializeJson(std::istream &is)
 {
 	Json::Value root;
 	is >> root;
-	if (root.isObject()) {
-		if (root["full_punch_interval"].isDouble())
-			full_punch_interval = root["full_punch_interval"].asFloat();
-		if (root["max_drop_level"].isInt())
-			max_drop_level = root["max_drop_level"].asInt();
-		if (root["punch_attack_uses"].isInt())
-			punch_attack_uses = root["punch_attack_uses"].asInt();
-
-		Json::Value &groupcaps_object = root["groupcaps"];
-		if (groupcaps_object.isObject()) {
-			Json::ValueIterator gciter;
-			for (gciter = groupcaps_object.begin();
-					gciter != groupcaps_object.end(); ++gciter) {
-				ToolGroupCap groupcap;
-				groupcap.fromJson(*gciter);
-				groupcaps[gciter.key().asString()] = groupcap;
-			}
-		}
 
-		Json::Value &damage_groups_object = root["damage_groups"];
-		if (damage_groups_object.isObject()) {
-			Json::ValueIterator dgiter;
-			for (dgiter = damage_groups_object.begin();
-					dgiter != damage_groups_object.end(); ++dgiter) {
-				Json::Value &value = *dgiter;
-				if (value.isInt())
-					damageGroups[dgiter.key().asString()] =
-						value.asInt();
-			}
-		}
+	if (!root.isObject())
+		return;
+
+	if (root["full_punch_interval"].isDouble())
+		full_punch_interval = root["full_punch_interval"].asFloat();
+
+	if (root["max_drop_level"].isInt())
+		max_drop_level = root["max_drop_level"].asInt();
+
+	if (root["punch_attack_uses"].isInt())
+		punch_attack_uses = root["punch_attack_uses"].asInt();
+
+	deserializeJsonGroupcaps(root["groupcaps"]);
+	deserializeJsonDamageGroups(root["damage_groups"]);
+}
+
+void ToolCapabilities::deserializeJsonGroupcaps(Json::Value &object) {
```

Comment:
This parameter is named `json` in other places, so let's keep it consistent. I think `json` is more descriptive anyway.